lang/java/avro/src/test/java/org/apache/avro/reflect/TestReflectData.java

```diff
@@ -130,4 +130,34 @@ private void validateSchema(Meta meta) {
       assertEquals("Invalid field " + field.name(), field.defaultVal(), testCases.get(field.name()));
     }
   }
+
+  public class Definition {
+    public Map<String, String> tokens;
+  }
+
+  @Test
+  public void testInnerClasses() {
+    testStaticInnerClasses();
+    testNonStaticInnerClasses();
+  }
+
+  public void testNonStaticInnerClasses() {
+    boolean successful = false;
+    try {
+      ReflectData.get().getSchema(Definition.class);
+    } catch (AvroTypeException ex) {
+      if (ex.getMessage().contains("must be a static inner class")) {
+        successful = true;
+      }
+    }
+    assertTrue(successful);
+  }
+
+  public void testStaticInnerClasses() {
```

Comment:
Same here, I'd love to use the annotation, and remove the fail("") part. If it throws an exception, it will fail anyway

Reply:
@Fokko code is fixed :D